tests/test_daemon_query.py

```
@@ -0,0 +1,284 @@
+
+            # Should exit with error code 1
+            assert exc_info.value.code == 1
+
+            # Verify query_daemon was NOT called
+            mock_query_func.assert_not_called()
+
+            # Verify error message
+            captured = capsys.readouterr()
+            assert "Error: either CMD or --json must be provided" in captured.err
+
+    def test_query_json_precedence_over_cmd(self, capsys):
```

Ruff (0.14.11)

Unused method argument: `capsys`

(ARG002)